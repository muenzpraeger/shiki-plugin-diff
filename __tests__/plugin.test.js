const shiki = require('shiki')
const diffPlugin = require('../index')

test('Default configuration', async () => {
  const highlighter = await shiki.getHighlighter({
    theme: 'github-light',
    langs: ['js'],
  })

  const code = `// This is an example of a diff
+  function Blog({ posts }) {
-  function Blog({ entries }) {
+    if (posts.length === 0) {
-    if (entries.length === 0) {
        return <p>No posts</p> 
     }
   }
`

  highlighter.usePlugin(diffPlugin)

  const out = highlighter.codeToHtml(code, { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Custom style colors', async () => {
  const highlighter = await shiki.getHighlighter({
    theme: 'github-light',
    langs: ['js'],
  })

  const code = `// This is an example of a diff
+  function Blog({ posts }) {
-  function Blog({ entries }) {
+    if (posts.length === 0) {
-    if (entries.length === 0) {
        return <p>No posts</p> 
     }
   }
`

  highlighter.usePlugin(diffPlugin({ colorAdd: 'blue', colorRemove: 'orange' }))

  const out = highlighter.codeToHtml(code, { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Theme colors for styles', async () => {
  const highlighter = await shiki.getHighlighter({
    theme: 'github-light',
    langs: ['js'],
  })

  const code = `// This is an example of a diff
+  function Blog({ posts }) {
-  function Blog({ entries }) {
+    if (posts.length === 0) {
-    if (entries.length === 0) {
        return <p>No posts</p> 
     }
   }
`

  highlighter.usePlugin(
    diffPlugin({
      applyThemeColorsForStyles: true,
    })
  )

  const out = highlighter.codeToHtml(code, { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Default classes instead of styles', async () => {
  const highlighter = await shiki.getHighlighter({
    theme: 'github-light',
    langs: ['js'],
  })

  const code = `// This is an example of a diff
+  function Blog({ posts }) {
-  function Blog({ entries }) {
+    if (posts.length === 0) {
-    if (entries.length === 0) {
        return <p>No posts</p> 
     }
   }
`

  highlighter.usePlugin(
    diffPlugin({
      useCssClasses: true,
    })
  )

  const out = highlighter.codeToHtml(code, { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Custom classes', async () => {
  const highlighter = await shiki.getHighlighter({
    theme: 'github-light',
    langs: ['js'],
  })

  const code = `// This is an example of a diff
+  function Blog({ posts }) {
-  function Blog({ entries }) {
+    if (posts.length === 0) {
-    if (entries.length === 0) {
        return <p>No posts</p> 
     }
   }
`

  highlighter.usePlugin(
    diffPlugin({
      useCssClasses: true,
      classAdd: 'add-me',
      classRemove: 'remove-me',
    })
  )

  const out = highlighter.codeToHtml(code, { lang: 'js' })

  expect(out).toMatchSnapshot()
})

test('Line numbers instead of RegEx', async () => {
  const highlighter = await shiki.getHighlighter({
    theme: 'github-light',
    langs: ['js'],
  })

  const code = `// This is an example of a diff
function Blog({ posts }) {
function Blog({ entries }) {
  if (posts.length === 0) {
  if (entries.length === 0) {
     return <p>No posts</p> 
  }
}
`

  highlighter.usePlugin(
    diffPlugin({
      useLineNumbers: true,
      linesAdd: [2, 4],
      linesRemove: [3, 5],
    })
  )

  const out = highlighter.codeToHtml(code, { lang: 'js' })

  expect(out).toMatchSnapshot()
})
