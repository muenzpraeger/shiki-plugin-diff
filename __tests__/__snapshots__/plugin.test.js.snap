// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom classes 1`] = `
"<pre class="shiki github-light" style="background-color: #ffffff;"><code><span class="line"><span style="color: #6E7781;">// This is an example of a diff</span></span>
<span class="line add-me"><span style="color: #CF222E;">+</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">posts</span><span style="color: #24292F;"> }) {</span></span>
<span class="line remove-me"><span style="color: #CF222E;">-</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">entries</span><span style="color: #24292F;"> }) {</span></span>
<span class="line add-me"><span style="color: #CF222E;">+</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (posts.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line remove-me"><span style="color: #CF222E;">-</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (entries.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line"><span style="color: #24292F;">        </span><span style="color: #CF222E;">return</span><span style="color: #24292F;"> &lt;</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt;No posts&lt;/</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt; </span></span>
<span class="line"><span style="color: #24292F;">     }</span></span>
<span class="line"><span style="color: #24292F;">   }</span></span>
<span class="line"></span></code></pre>"
`;

exports[`Custom style colors 1`] = `
"<pre class="shiki github-light" style="background-color: #ffffff;"><code><span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #6E7781;">// This is an example of a diff</span></span>
<span class="line" style="background-color: blue; display: inline-block; width: 100%;"><span style="color: #CF222E;">+</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">posts</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: orange; display: inline-block; width: 100%;"><span style="color: #CF222E;">-</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">entries</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: blue; display: inline-block; width: 100%;"><span style="color: #CF222E;">+</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (posts.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: orange; display: inline-block; width: 100%;"><span style="color: #CF222E;">-</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (entries.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">        </span><span style="color: #CF222E;">return</span><span style="color: #24292F;"> &lt;</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt;No posts&lt;/</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt; </span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">     }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">   }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"></span></code></pre>"
`;

exports[`Default classes instead of styles 1`] = `
"<pre class="shiki github-light" style="background-color: #ffffff;"><code><span class="line"><span style="color: #6E7781;">// This is an example of a diff</span></span>
<span class="line diff-add"><span style="color: #CF222E;">+</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">posts</span><span style="color: #24292F;"> }) {</span></span>
<span class="line diff-remove"><span style="color: #CF222E;">-</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">entries</span><span style="color: #24292F;"> }) {</span></span>
<span class="line diff-add"><span style="color: #CF222E;">+</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (posts.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line diff-remove"><span style="color: #CF222E;">-</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (entries.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line"><span style="color: #24292F;">        </span><span style="color: #CF222E;">return</span><span style="color: #24292F;"> &lt;</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt;No posts&lt;/</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt; </span></span>
<span class="line"><span style="color: #24292F;">     }</span></span>
<span class="line"><span style="color: #24292F;">   }</span></span>
<span class="line"></span></code></pre>"
`;

exports[`Default configuration 1`] = `
"<pre class="shiki github-light" style="background-color: #ffffff;"><code><span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #6E7781;">// This is an example of a diff</span></span>
<span class="line" style="background-color: #00ff4051; display: inline-block; width: 100%;"><span style="color: #CF222E;">+</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">posts</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: #ff000051; display: inline-block; width: 100%;"><span style="color: #CF222E;">-</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">entries</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: #00ff4051; display: inline-block; width: 100%;"><span style="color: #CF222E;">+</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (posts.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: #ff000051; display: inline-block; width: 100%;"><span style="color: #CF222E;">-</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (entries.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">        </span><span style="color: #CF222E;">return</span><span style="color: #24292F;"> &lt;</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt;No posts&lt;/</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt; </span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">     }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">   }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"></span></code></pre>"
`;

exports[`Line numbers instead of RegEx 1`] = `
"<pre class="shiki github-light" style="background-color: #ffffff;"><code><span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #6E7781;">// This is an example of a diff</span></span>
<span class="line" style="background-color: #00ff4051; display: inline-block; width: 100%;"><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">posts</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: #ff000051; display: inline-block; width: 100%;"><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">entries</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: #00ff4051; display: inline-block; width: 100%;"><span style="color: #24292F;">  </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (posts.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: #ff000051; display: inline-block; width: 100%;"><span style="color: #24292F;">  </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (entries.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">     </span><span style="color: #CF222E;">return</span><span style="color: #24292F;"> &lt;</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt;No posts&lt;/</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt; </span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">  }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">}</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"></span></code></pre>"
`;

exports[`Theme colors for styles 1`] = `
"<pre class="shiki github-light" style="background-color: #ffffff;"><code><span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #6E7781;">// This is an example of a diff</span></span>
<span class="line" style="background-color: #6fdd8b66; display: inline-block; width: 100%;"><span style="color: #CF222E;">+</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">posts</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: #ffaba866; display: inline-block; width: 100%;"><span style="color: #CF222E;">-</span><span style="color: #24292F;">  </span><span style="color: #CF222E;">function</span><span style="color: #24292F;"> </span><span style="color: #8250DF;">Blog</span><span style="color: #24292F;">({ </span><span style="color: #953800;">entries</span><span style="color: #24292F;"> }) {</span></span>
<span class="line" style="background-color: #6fdd8b66; display: inline-block; width: 100%;"><span style="color: #CF222E;">+</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (posts.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: #ffaba866; display: inline-block; width: 100%;"><span style="color: #CF222E;">-</span><span style="color: #24292F;">    </span><span style="color: #CF222E;">if</span><span style="color: #24292F;"> (entries.</span><span style="color: #0550AE;">length</span><span style="color: #24292F;"> </span><span style="color: #CF222E;">===</span><span style="color: #24292F;"> </span><span style="color: #0550AE;">0</span><span style="color: #24292F;">) {</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">        </span><span style="color: #CF222E;">return</span><span style="color: #24292F;"> &lt;</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt;No posts&lt;/</span><span style="color: #116329;">p</span><span style="color: #24292F;">&gt; </span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">     }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"><span style="color: #24292F;">   }</span></span>
<span class="line" style="background-color: transparent; display: inline-block; width: 100%;"></span></code></pre>"
`;
